Source/AwsCommonRuntimeKit/auth/credentials/CRTAWSCredentialsProvider.swift

```diff
@@ -4,51 +4,109 @@
 import AwsCAuth
 import AwsCIo
 import AwsCHttp
+import Foundation
 
-public final class CRTAWSCredentialsProvider {
+public protocol CredentialsProvider: AnyObject {
+    func getCredentials() async throws -> AwsCredentials
+}
 
-    let allocator: Allocator
+/// A container class to wrap a protocol so that we use it with Unmanaged
+class DelegateCredentialsProviderCore {
```

Comment:
What does the word _delegate_ mean in this context? Historically in Swift (and objc before that), the word _delegate_ was used as a noun and referred to the [delegation pattern](https://developer.apple.com/documentation/swift/using-delegates-to-customize-object-behavior).

It is a bit unexpected and confusing to see an object named with `Delegate` at the front.
If this is used just for the `AwsCredentialsProvider` below can we embed it within the `AwsCredentialsProvider` and just name it `Core`? If this is a common pattern in CRT, we can do this elsewhere as well and it should simplify naming while also making the intended usage more clear.

```swift
public class AWSCredentialsProvider {
    class Core {
       // ...
    }
}

```

Reply:
Thanks, making it an inner class is good idea. AWSCredentialsProvider has a lot of different types. One of them is a delegate credential provider which delegates fetching the credentials to the provided Credential Provider. We need to do some manual reference counting to keep it alive until the underlying C credential provider is alive.